_posts/2023-11-15-Codesgiving-Part-1.md

```diff
@@ -0,0 +1,289 @@
+
+> ...
+> Simple is better than complex.
+> Complex is better than complicated.
+> ...
+> \- [PEP 20: The Zen of Python](https://peps.python.org/pep-0020/)
+
+Further channeling step 3. from test-driven development above, prefer simple changes over more complex ones (recognizing that the _absolute_ simplest can take iteration and thought).
+Some of the best solutions are often the most easily understood ones (where the code addition or changes seem obvious afterwards).
+It might be tempting to address more than one bug or feature at the same time.
+___Avoid [feature creep](https://en.wikipedia.org/wiki/Feature_creep) as you build solutions - stay focused on the task at hand!___
+Take note of things you notice on your journey to address the reported needs.
+These can be become additional reported bugs or features which could be addressed later.
+Staying focused with your development will save you time, keep your tests constrained, and (theoretically) help reduce the time and complexity of code review.
+
+### Developing a Solution
+
+Once you have a test in place for the bug fix or feature addition it's time to work towards developing a solution.
+If you've taken time to accomplish the prior steps before this point you may already have a good idea about how to go about a solution.
+If not, spend some time investigating the technical aspects of a solution, optionally adding this information to the report or discussion content for further review before development.
+
+Use [timeboxing techniques](https://cu-dbmi.github.io/set-website/2023/01/17/Timebox-Your-Software-Work.html) to  help make sure the time you spend in development is no more than necessary.
+Also recall that a "simplest version" of the code can often be more quickly refactored and completed than devising a "perfect" solution the first time.
+Remember, you'll also very likely have the help of a code review before the code is merged (expect to learn more and add changes during review!).
+
+## Code Review, Revisions, and Post-actions
+
+### Pull Requests and Code Review
+
+When your code and new test(s) are in a good spot it's time to ask for a code review.
+Be sure to follow the open-source [`CONTRIBUTING.md` document](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) documentation related to the project; each one can vary.
+When working on GitHub-based projects, you'll need to open a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) on the correct branch (usually upstream `main`).
+If you used a GitHub issue to help report the issue, mention the issue in the pull request description using the `#<issue number` reference to help link the work to the reported need (this will cause the pull request to show up within the issue).
+
+Note: It might feel tempting to perfect the code.
```

Comment:
I feel like this point is said several times in article. Maybe consider reviewing if any of the instances can be removed. Though it is certainly a good thing to emphasize multiple times, and keep in mind at all times during the process.